airbyte-cdk/python/airbyte_cdk/sources/streams/http/http_client.py

```diff
@@ -250,14 +251,28 @@ def _send(
                 "Receiving response", extra={"headers": response.headers, "status": response.status_code, "body": response.text}
             )
 
-        # Request/repsonse logging for declarative cdk.
+        # Request/response logging for declarative cdk
         if log_formatter is not None and response is not None and self._message_repository is not None:
             formatter = log_formatter
             self._message_repository.log_message(
                 Level.DEBUG,
                 lambda: formatter(response),  # type: ignore # log_formatter is always cast to a callable
             )
 
+        # Emit stream status RUNNING with the reason RATE_LIMITED to log that the rate limit has been reached
+        if error_resolution.response_action == ResponseAction.RATE_LIMITED:
+            # TODO: Update to handle with message repository when concurrent message repository is ready
+            reasons = [AirbyteStreamStatusReason(type=AirbyteStreamStatusReasonType.RATE_LIMITED)]
+            message = stream_status_as_airbyte_message(StreamDescriptor(name=self._name), AirbyteStreamStatus.RUNNING, reasons).json(
+                exclude_unset=True
+            )
+
+            # Simply printing the stream status is a temporary solution and can cause future issues. Currently, the _send method is
+            # wrapped with backoff decorators, and we can only emit messages by iterating record_iterator in the abstract source at the
+            # end of the retry decorator behavior. This approach does not allow us to emit messages in the queue before exiting the
+            # backoff retry loop.
+            print(message)
```

Comment:
just to verify, this has been tested and prints as expected on a test sync?

Reply:
yes
<img width="1549" alt="image" src="https://github.com/airbytehq/airbyte/assets/53845333/d1f9a9bc-0f20-4c22-a030-5b1ccfc7acf4">